addons/purchase/models/purchase.py

```diff
@@ -722,7 +722,9 @@ def _prepare_stock_moves(self, picking):
         for move in self.move_ids.filtered(lambda x: x.state != 'cancel' and not x.location_dest_id.usage == "supplier"):
             qty += move.product_uom._compute_quantity(move.product_uom_qty, self.product_uom, rounding_method='HALF-UP')
         template = {
-            'name': self.name or '',
+            # truncate to 2000 to avoid triggering index limit error
+            # TODO: remove index in master?
+            'name': self.name[:2000] or '',
```

Comment:
Although the `name` is required, it's probably safer to do:

```suggestion
            'name': (self.name or '')[:2000],
```